application/src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -1765,6 +1808,50 @@ type RealTimeEstimate {
   time: OffsetDateTime!
 }
 
+"""
+Timing of an arrival or a departure to or from a stop. May contain real-time information if
+available. This is used when there is a known scheduled time.
+"""
+type RegularRealTimeArrivalDepartureTime {
```

Comment:
I am not convinced this is a good name. 
- Since this is not part of a hierarchy, the "Regular" should be dropped.
- I suggested `RealTime`, but we are using this without real-time as well. It is normally ok to name something for the "main" feature and skip supporting features, but in this case I think it is stretching it.

We need a term for "visiting a stop". The best I can think of is `Call`. In Siri we have `EstimatedCall` and `ExtraCall`, GTFS does not have an expression for this - so I think we should use the transmodel expression. 

If we go for call, than this should be `CallTime` and it should possibly be extended with _actual/recorded time_ later. If you insist of using `StopTime`, then it would suggest `EstimatedStopTime`, but this does not contain the stop so I think it is wrong.

Reply:
This regular vs no prefix boils down to the discussions we have had recently on if all different options should all have a prefix or is it ok that the "standard/regular" one doesn't have it but the others do. As for `RealTime`, we might have a version of this type in the future for fetching the planned schedules which doesn't ever have an option of having realtime data. However, if we just have the scheduled time, do we even need a wrapper class at all? But in theory, I can image that in the future we would have at least these four versions:
1. "Regular RealTime", can contain estimates, times are exact, not windows. Do we call this `RegularRealTimeX`, `RealTimeX`, `Regular` or just `X`?
3. "Regular" can't contain estimates, times are exact, not windows. Do we call this `RegularScheduledX`, `ScheduledX`, `RegularX` or just `X`?
4. "Flex RealTime",  contains estimates, estimates are probably exact times while scheduled time is a window. Do we call this `FlexRealTimeX` or`FlexX`?
5. "Flex" can't contain estimates, times are windows. Do we call this `FlexScheduledX` or`FlexX`?

Reply:
I've now changed them to use the shortest naming convention without Regular or RealTime suffix, but we can change that still if we thought those names should be reserved for something else.

Reply:
A `CallTime` has a scheduled time, an optional estimated time and an optional actual time.

To support FLEX we need to add time-window to the above information, the actual time and estimates can be the same. There are several ways to do it. In my view the best way to do it is to have @oneOf inside `CallTime` for `scheduledTime` or `scheduledWindow`. You could also have `ScheduledWindowCallTime` and `ScheduledExactCallTime` and an interface `CallTime`. I do not like this, because you let one attribute dictate the type - you create two types for the same thing - duplicating almost everything inside. If another feature of the call have the same split, then you end up with 4 types. Note! Even if regular transit only uses  `ScheduledExactCallTime`, Fles uses both of these, so there is no such thing as a `FlexCallTime`.